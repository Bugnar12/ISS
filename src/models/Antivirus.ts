export class Antivirus{
    private _id: number;
    private _name: string;
    private _producer: string;
    private _description: string;
    private _supportMultiPlatform: boolean;
    private _releaseDate: Date;

    public constructor(id: number, name: string, producer: string, description: string, supportMultiPlatform: boolean, releaseDate: Date) {
        this._id = id;
        this._name = name;
        this._producer = producer;
        this._description = description;
        this._supportMultiPlatform = supportMultiPlatform;
        this._releaseDate = releaseDate;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get producer(): string {
        return this._producer;
    }

    set producer(value: string) {
        this._producer = value;
    }

    get description(): string {
        return this._description;
    }

    set description(value: string) {
        this._description = value;
    }

    get supportMultiPlatform(): boolean {
        return this._supportMultiPlatform;
    }

    set supportMultiPlatform(value: boolean) {
        this._supportMultiPlatform = value;
    }

    get releaseDate(): Date {
        return this._releaseDate;
    }

    set releaseDate(value: Date) {
        this._releaseDate = value;
    }
}