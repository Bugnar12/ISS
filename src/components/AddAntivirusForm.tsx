import React, {useState} from 'react';

import {Antivirus} from "../models/Antivirus";
import {addAntivirus} from "../api/API_Requests";
import "../components/controls/StyleInput.css";

interface Props{
    onSubmit: (antivirus: Antivirus) => void;
}

const AddAntivirusForm: React.FC<Props>= ({onSubmit}) => {
    const [id, setId] = useState(0);
    const [name, setName] = useState('');
    const [producer, setProducer] = useState('');
    const [description, setDescription] = useState('');
    const [supportMultiPlatform, setSupportMultiPlatform] = useState(false);
    const [releaseDate, setReleaseDate] = useState('');

    const handleSubmit = (event: React.FormEvent) => {
        event.preventDefault();
        if (!name || !producer || !description || !releaseDate) {
            alert('Please fill out all fields');
            return;
        }

        const newId = Date.now();
        const newAntivirus = new Antivirus(newId, name, producer, description, supportMultiPlatform, new Date(releaseDate));

        const antivirusPlainObject = {
            id: newId,
            name: name,
            producer: producer,
            description: description,
            supportMultiPlatform: supportMultiPlatform,
            releaseDate: new Date(releaseDate)
        }

        let antiviruses: typeof antivirusPlainObject[] = [];
        const offlineAntiviruses = localStorage.getItem('offlineAntiviruses');
        if (offlineAntiviruses) {
            antiviruses = JSON.parse(offlineAntiviruses);
        }

        if (navigator.onLine) {
            // If the user is online, make the API request
            addAntivirus(antivirusPlainObject)
                .then(() => {
                    onSubmit(newAntivirus);
                })
                .catch(error => {
                    console.error('Error adding antivirus: ', error);
                });
        } else {
            console.log('intru aici');
            console.log(antivirusPlainObject)
            // If the user is offline, store the data in localStorage
            antiviruses.push(antivirusPlainObject);
            localStorage.setItem('offlineAntiviruses', JSON.stringify(antiviruses));
        }
    };

    return(
        <div>
            <form onSubmit={handleSubmit}>
                <input type="text" className="add-input-style" value={name} onChange={e => setName(e.target.value)} placeholder="Name"/>
                <input type="text" className="add-input-style" value={producer} onChange={e => setProducer(e.target.value)} placeholder="Producer"/>
                <input type="text" className="add-input-style" value={description} onChange={e => setDescription(e.target.value)}
                       placeholder="Description"/>
                <input type="checkbox" className="add-input-style" checked={supportMultiPlatform}
                       onChange={e => setSupportMultiPlatform(e.target.checked)}/>Multi-Platform Support
                <input type="date" className="add-input-style" value={releaseDate} onChange={e => setReleaseDate(e.target.value)} placeholder="Release Date"/>
                <input type="submit" value="Submit" />
            </form>
        </div>
    )
}

export default AddAntivirusForm;