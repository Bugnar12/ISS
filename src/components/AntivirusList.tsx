import {Antivirus} from "../models/Antivirus";
import '../App.css'

export const antivirusList : Antivirus[] = [
    new Antivirus(1, 'Kaspersky', 'Kaspersky Lab', 'Kaspersky is a well-known antivirus software', true, new Date('2020-01-01')),
    new Antivirus(2, 'Bitdefender', 'Bitdefender', 'Bitdefender is a romanian antivirus.', true, new Date('2001-11-06')),
    new Antivirus(3, 'Avast', 'Avast Software', 'Avast is a czech antivirus software', true, new Date('2010-10-10')),
    new Antivirus(4, 'Norton', 'NortonLifeLock', 'Norton is an antivirus software', true, new Date('2016-01-01')),
    new Antivirus(5, 'McAfee', 'McAfee', 'McAfee is an antivirus software', true, new Date('2006-01-01')),
    new Antivirus(6, 'ESET', 'ESET', 'ESET is an antivirus software', true, new Date('2017-01-01')),
    new Antivirus(7, 'SuperAntiSpyware', 'SuperAntiSpyware', 'SuperAntiSpyware is an antivirus software', false, new Date('2000-01-01')),
    new Antivirus(8, 'Malwarebytes', 'Malwarebytes', 'Malwarebytes is an antivirus software', true, new Date('2018-01-01')),
    new Antivirus(9, 'Spybot', 'Safer-Networking', 'Spybot is an antivirus software', false, new Date('2002-01-01')),
    new Antivirus(10, 'Windows Defender', 'Microsoft', 'Windows Defender is an antivirus software', true, new Date('2015-01-01')),
    new Antivirus(11, 'Avira', 'Avira', 'Avira is an antivirus software', true, new Date('2019-01-01')),
    new Antivirus(12, 'Panda', 'Panda Security', 'Panda is an antivirus software', true, new Date('2011-01-01')),
    new Antivirus(13, 'AVG', 'AVG Technologies', 'AVG is an antivirus software', true, new Date('2008-01-01')),
    new Antivirus(14, 'WhatAVirus', 'WhatAVirus', 'WhatAVirus is an antivirus software', false, new Date('2017-01-01')),
    new Antivirus(15, 'VirusDestroyer', 'VirusDestroyer', 'VirusDestroyer is an antivirus software', false, new Date('2000-01-01')),
    new Antivirus(16, 'AntivirusX', 'AntivirusX', 'AntivirusX is an antivirus software', false, new Date('2000-06-01')),
    new Antivirus(17, 'AntivirusY', 'AntivirusY', 'AntivirusY is an antivirus software', false, new Date('2000-01-02')),
    new Antivirus(18, 'AntivirusZ', 'AntivirusZ', 'AntivirusZ is an antivirus software', false, new Date('2014-03-06')),
    new Antivirus(19, 'BestAntivirus', 'BestAntivirus', 'BestAntivirus is an antivirus software', false, new Date('2000-01-01')),
    new Antivirus(20, 'WorstAntivirus', 'WorstAntivirus', 'WorstAntivirus is an antivirus software', false, new Date('2000-01-01')),
    new Antivirus(21, 'Antivirus2000', 'Antivirus2000', 'Antivirus2000 is an antivirus software', false, new Date('2000-01-01')),
    new Antivirus(22, 'Antivirus2001', 'Antivirus2001', 'Antivirus2001 is an antivirus software', false, new Date('2001-01-01')),
    new Antivirus(23, 'Antivirus2002', 'Antivirus2002', 'Antivirus2002 is an antivirus software', false, new Date('2002-01-01')),
    new Antivirus(24, 'Antivirus2003', 'Antivirus2003', 'Antivirus2003 is an antivirus software', false, new Date('2003-01-01')),
    new Antivirus(25, 'FlowerAntivirus', 'FlowerAntivirus', 'FlowerAntivirus is an antivirus software', false, new Date('2000-01-01')),
    new Antivirus(26, 'Antivirus2020', 'Antivirus2020', 'Antivirus2020 is an antivirus software', false, new Date('2020-01-01')),
    new Antivirus(27, 'HarmlessAntivirus', 'HarmlessAntivirus', 'HarmlessAntivirus is an antivirus software', false, new Date('2000-01-01')),
    new Antivirus(28, 'MyAntivirus', 'MyAntivirus', 'MyAntivirus is an antivirus software', false, new Date('2000-01-01')),
]
