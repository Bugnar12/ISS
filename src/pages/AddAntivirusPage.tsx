// src/pages/AddAntivirusPage.tsx
import React from 'react';

import { Antivirus } from '../models/Antivirus';
import AddAntivirusForm from '../components/AddAntivirusForm';
import {useNavigate} from "react-router-dom";

const AddAntivirusPage: React.FC = () => {
    const navigate = useNavigate();

    const handleSubmit = (antivirus: Antivirus) => {
        navigate('/');
    };

    return (
        <div>
            <AddAntivirusForm onSubmit={handleSubmit} />
        </div>
    );
};

export default AddAntivirusPage;