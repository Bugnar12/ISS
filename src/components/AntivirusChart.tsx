import { PieChart, pieArcLabelClasses } from '@mui/x-charts/PieChart';
import { antivirusList } from "./AntivirusList";

const sizing = {
    margin: { left: 5 },
    width: 400,
    height: 350,
};

export default function AntivirusChart() {
    const data = antivirusList.map(antivirus => {
        return {
            label: antivirus.name,
            value: antivirus.releaseDate.getFullYear() > 2015 ? 1 : 0
        }
    });

    return (
        <div className="chart-container" style={{ width: 400, height: 400, }}>
            <PieChart
                series={[
                    {
                        outerRadius: 90,
                        data,
                    },
                ]}
            sx={{
                [`& .${pieArcLabelClasses.root}`]:{
                    fill: 'white',
                    fonSize: 14,
                },
            }}
            slotProps={{
                legend: {
                    //hide the legend
                    hidden: true,
            },
            }}
            {...sizing}
        />
        </div>
    );
}