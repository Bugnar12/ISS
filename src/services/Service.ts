import {Antivirus} from '../models/Antivirus'
import {antivirusList} from '../components/AntivirusList'

//this array will be used to store the data

/**
 * Adds a new antivirus to the antivirusList array.
 *
 * @param {Antivirus} antivirus - The antivirus object to be added to the list.
 * The function first calculates the maximum id from the existing antivirus objects in the list,
 * increments it by one and assigns it to the new antivirus object.
 */
export function addAntivirus(antivirus: Antivirus)
{
    const addedID = Math.max(...antivirusList.map(e => e.id), 0) + 1;
    antivirus.id = addedID;
    antivirusList.push(antivirus);
    console.log(addedID);
}