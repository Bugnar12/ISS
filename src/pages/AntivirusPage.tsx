import React, { useState, useEffect, useCallback } from 'react';
import Modal from 'react-modal';
import AntivirusList2 from "../components/AntivirusList2";
import DetailView from "../components/DetailView";
import {Antivirus} from "../models/Antivirus";
import {Link} from "react-router-dom";
import DeleteButton from "../components/controls/DeleteButton";
import '../components/controls/ReactModal.css';
import {deleteAntivirus, getAntivirusListPaging, getAntivirusList} from "../services/Service";
import AntivirusChart from "../components/AntivirusChart";
import "../components/ChartStyle.css"
import axios from "axios";


//Modal.setAppElement('#root'); // This line is needed for accessibility reasons

const AntivirusPage: React.FC = () => {
    const [selectedAntivirus, setObject] = useState<Antivirus | null>(null);
    const [antivirusList, setAntivirusList] = useState<Antivirus[]>([]);
    const [isAscending, setIsAscending] = useState<boolean>(true);
    const [modalIsOpen, setModalIsOpen] = useState(false);
    const [currentPage, setCurrentPage] = useState(1);
    const itemsPerPage = 5;

    const handleNextPage = () =>{
        if(currentPage < Math.ceil(antivirusList.length / itemsPerPage))
            setCurrentPage(currentPage + 1);
    }

    const handlePreviousPage = () => {
        if(currentPage > 1)
        {
            setCurrentPage(currentPage - 1);
        }
    }

    const paginatedAntivirusList = antivirusList.slice((currentPage - 1) * itemsPerPage, currentPage * itemsPerPage);


    useEffect(() => {
        const antivirusesList = getAntivirusList();
        setAntivirusList(antivirusesList);

    }, []);

    const handleDelete = () => {
        setModalIsOpen(true);
    };

    const handleConfirmDelete = () => {
        //delete + filter to see which id has the antivirus selected + set the object to null
        if (selectedAntivirus) {
            deleteAntivirus(selectedAntivirus);
            setAntivirusList(antivirusList.filter(antivirus => antivirus !== selectedAntivirus));
            setObject(null);
        }
        setModalIsOpen(false);
    };

    const handleCancelDelete = () => {
        setModalIsOpen(false);
    };

    const handleClickEvent = (antivirus: Antivirus) => {
        if (selectedAntivirus && selectedAntivirus.id === antivirus.id) {
            setObject(null);
        } else {
            setObject(antivirus);
        }
    };

    const handleSort = () =>{
        setIsAscending(!isAscending);
        antivirusList.sort((a, b) => {
          if(isAscending)
          {
              return a.name.localeCompare(b.name);
          }
          else
              return b.name.localeCompare(a.name);
        })
    }


    return(
        <div className="antivirus-page">
            <div className="button-list-container">
                <div className="style-multiple-buttons">
                    <div className="styled-button">
                        <Link to="/add" className="style-link">Add Antivirus</Link>
                    </div>
                    <DeleteButton antivirus={selectedAntivirus} onDelete={handleDelete}/>
                    <div className="styled-button">
                        <Link to={`/update/${selectedAntivirus?.id}`} className={"style-link"}>Update Antivirus</Link>
                    </div>
                    <div>
                        <button onClick={handleSort} className="styled-button">Sorting</button>
                    </div>
                </div>
                <div className="list-container">
                    <AntivirusList2 antivirusList={paginatedAntivirusList} onAntivirusClick={handleClickEvent}
                                    selectedAntivirus={selectedAntivirus}/>
                    <button onClick={handlePreviousPage}>Previous Page</button>
                    <button onClick={handleNextPage}>Next Page</button>
                </div>
            </div>
            <div className="wrapper-detail-chart">
                <div className="detail-container">
                    <DetailView antivirus={selectedAntivirus}/>
                </div>
                <div className="box">
                    <div>
                        <b>Release date after 12-31-2015: </b>
                    </div>
                    <AntivirusChart/>
                </div>
            </div>
            <Modal className="react-modal"
                   isOpen={modalIsOpen}
                   onRequestClose={handleCancelDelete}
                   contentLabel="Delete Confirmation"
            >
                <h2>Confirm Delete</h2>
                <p>Are you sure you want to delete this antivirus?</p>
                <button onClick={handleConfirmDelete}>Yes</button>
                <button onClick={handleCancelDelete}>No</button>
            </Modal>
        </div>
    )
};
export default AntivirusPage;