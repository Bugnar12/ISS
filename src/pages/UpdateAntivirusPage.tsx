import React, { useEffect, useState } from 'react';
import {useNavigate, useParams} from "react-router-dom";
import {Antivirus} from "../models/Antivirus";
import {updateAntivirus, getAntivirusList} from "../services/Service";
import UpdateAntivirusForm from "../components/UpdateAntivirusForm";

interface RouteParams {
    [key: string]: string;
}

const UpdateAntivirusPage: React.FC = () => {
    const navigate = useNavigate();
    const { antivirusIDString } = useParams<RouteParams>();
    const [antivirus, setAntivirus] = useState<Antivirus | null>(null);

    useEffect(() => {
        if (antivirusIDString) {
            const antivirusID = parseInt(antivirusIDString);
            const antivirusToUpdate = getAntivirusList().find(antivirus => antivirus.id === antivirusID);
            setAntivirus(antivirusToUpdate ? antivirusToUpdate : null);
        } else {
            // Handle the case when antivirusIDString is undefined
            // For example, you can redirect the user or show an error message
            navigate('/');
        }
    }, [antivirusIDString, navigate]);

    const handleUpdate = (antivirus: Antivirus) => {
        updateAntivirus(antivirus);
        navigate('/');
    };

    return(
        <div>
            <UpdateAntivirusForm onClick={handleUpdate} antivirus={antivirus}/>
        </div>
    )
}

export default UpdateAntivirusPage;