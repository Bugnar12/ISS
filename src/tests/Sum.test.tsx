import { addAntivirus, getAntivirusList, deleteAntivirus, updateAntivirus } from "../services/Service";
import { Antivirus } from "../models/Antivirus";
import {BrowserRouter as Router} from "react-router-dom";
import AntivirusPage from "../pages/AntivirusPage";
import UpdateAntivirusPage from "../pages/UpdateAntivirusPage";
import React from "react";
import renderer from 'react-test-renderer';
import AddAntivirusPage from "../pages/AddAntivirusPage";
import {antivirusList} from "../components/AntivirusList";
import exp from "node:constants";

// Test for Create operation
test('should add an antivirus to the list', () => {
    const antivirus = new Antivirus(8, "Test Antivirus", "Test Producer", "Test Description", true, new Date("2022-01-01"));
    addAntivirus(antivirus);
    expect(antivirus.id).toBe(8);
    expect(antivirusList.length).toBe(8);
    expect(antivirusList).toContain(antivirus);
});

// Test for Read operation
test('should get the list of antiviruses', () => {
    getAntivirusList();
    expect(getAntivirusList()).toEqual(antivirusList);
});

// Test for Update operation
test('should update an antivirus in the list', () => {
    const antivirus = new Antivirus(1, "Updated Antivirus", "Updated Producer", "Updated Description", true, new Date("2022-01-01"));
    updateAntivirus(antivirus);
    expect(antivirusList[0].name).toBe("Updated Antivirus");
    expect(antivirusList[0].producer).toBe("Updated Producer");
    expect(antivirusList[0].description).toBe("Updated Description");
    expect(antivirusList[0].supportMultiPlatform).toBe(true);
    expect(antivirusList[0].releaseDate).toEqual(new Date("2022-01-01"));
});

// Test for Delete operation
test('should delete an antivirus from the list', () => {
    const antivirus = new Antivirus(1, "Test Antivirus", "Test Producer", "Test Description", true, new Date("2022-01-01"));
    deleteAntivirus(antivirus);
    expect(antivirusList.length).toBe(7);
    expect(antivirusList).not.toContain(antivirus);
});

// Snapshot test for Create operation
test('should render AddAntivirusPage correctly', () => {
    const tree = renderer.create(<Router><AddAntivirusPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

// Snapshot test for Read operation
test('should render AntivirusPage correctly', () => {
    const tree = renderer.create(<Router><AntivirusPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

// Snapshot test for Update operation
test('should render UpdateAntivirusPage correctly', () => {
    const tree = renderer.create(<Router><UpdateAntivirusPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

// Snapshot test for Delete operation
// Since the delete operation does not have a specific page, we can test the AntivirusPage again as it includes the delete functionality
test('should render AntivirusPage correctly for delete operation', () => {
    const tree = renderer.create(<Router><AntivirusPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});