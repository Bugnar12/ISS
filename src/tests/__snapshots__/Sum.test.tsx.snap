// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddAntivirusPage correctly 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Producer"
        type="text"
        value=""
      />
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Description"
        type="text"
        value=""
      />
      <input
        checked={false}
        className="add-input-style"
        onChange={[Function]}
        type="checkbox"
      />
      Multi-Platform Support
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Release Date"
        type="date"
        value=""
      />
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`should render AntivirusPage correctly 1`] = `
<div
  className="antivirus-page"
>
  <div
    className="button-list-container"
  >
    <div
      className="style-multiple-buttons"
    >
      <div
        className="styled-button"
      >
        <a
          className="style-link"
          href="/add"
          onClick={[Function]}
        >
          Add Antivirus
        </a>
      </div>
      <button
        className="styled-button"
        onClick={[Function]}
      >
        Delete Antivirus
      </button>
      <div
        className="styled-button"
      >
        <a
          className="style-link"
          href="/update/undefined"
          onClick={[Function]}
        >
          Update Antivirus
        </a>
      </div>
      <div>
        <button
          className="styled-button"
          onClick={[Function]}
        >
          Sorting
        </button>
      </div>
    </div>
    <div
      className="list-container"
    >
      <ul
        className="style-list"
      />
    </div>
  </div>
  <div
    className="detail-container"
  />
</div>
`;

exports[`should render AntivirusPage correctly for delete operation 1`] = `
<div
  className="antivirus-page"
>
  <div
    className="button-list-container"
  >
    <div
      className="style-multiple-buttons"
    >
      <div
        className="styled-button"
      >
        <a
          className="style-link"
          href="/add"
          onClick={[Function]}
        >
          Add Antivirus
        </a>
      </div>
      <button
        className="styled-button"
        onClick={[Function]}
      >
        Delete Antivirus
      </button>
      <div
        className="styled-button"
      >
        <a
          className="style-link"
          href="/update/undefined"
          onClick={[Function]}
        >
          Update Antivirus
        </a>
      </div>
      <div>
        <button
          className="styled-button"
          onClick={[Function]}
        >
          Sorting
        </button>
      </div>
    </div>
    <div
      className="list-container"
    >
      <ul
        className="style-list"
      />
    </div>
  </div>
  <div
    className="detail-container"
  />
</div>
`;

exports[`should render UpdateAntivirusPage correctly 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Antivirus name:
      </label>
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
      <label>
        Producer
      </label>
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Producer"
        type="text"
        value=""
      />
      <label>
        Description
      </label>
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Description"
        type="text"
        value=""
      />
      <input
        checked={false}
        className="add-input-style"
        onChange={[Function]}
        type="checkbox"
      />
      Multi-Platform Support
      <input
        className="add-input-style"
        onChange={[Function]}
        placeholder="Release Date"
        type="date"
        value=""
      />
      <label>
        Release Date
      </label>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
